metastore/src/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -8444,6 +8558,143 @@ private String getPrimaryKeyConstraintName(String db_name, String tbl_name) thro
   }
 
   @Override
+  public List<SQLUniqueConstraint> getUniqueConstraints(String db_name, String tbl_name)
+          throws MetaException {
+    try {
+      return getUniqueConstraintsInternal(db_name, tbl_name, true, true);
+    } catch (NoSuchObjectException e) {
+      throw new MetaException(e.getMessage());
```

Comment:
instead of e.getMessage() use org.apache.commons.lang.exception.ExceptionUtils.getStackTrace(Throwable) to capture full stack trace.